app/src/lib/git/push.ts

```diff
@@ -53,6 +54,7 @@ export async function push(
   remote: IRemote,
   localBranch: string,
   remoteBranch: string | null,
+  tagsToPush: ReadonlyArray<string> | null,
```

Comment:
As opposed to what we talked @niik , I've ended up adding an independent argument to indicate the tags to push.

I've changed my mind because on L76 we set the `--set-upstream` argument when `remoteBranch` is `null`.

If we changed the signature of this method to accept a single array of tuples that would englobe both branches and tags (e.g `ReadonlyArray<[string, string | null]>` where the first element of the tuple is the local ref and the second one is the remote ref), then we wouldn't be able to know when exactly we should set the `--set-upstream` param (we could set it when the remote reference was `null`, but then it would mean setting `--set-upstream` every time we push tags).

Anyways, at first glance it doesn't seem like adding the `--set-upstream` arg when pushing tags could cause issues, but I don't think that the marginal improvement on cleanliness for this method is worth the risk